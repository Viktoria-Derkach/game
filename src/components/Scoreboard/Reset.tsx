import React, { FC, useState, memo } from 'react';
import styled from '@emotion/styled';

export interface ResetProps {
  /**
   * Reset action handler
   */
  onReset: () => void;
}

export const Reset: FC<ResetProps> = memo(({ onReset }) => {
  const [mouseDown, setMouseDown] = useState(false);

  const onMouseDown = () => setMouseDown(true);
  const onMouseUp = () => setMouseDown(false);
  console.log('defd');

  return (
    <Button
      onMouseDown={onMouseDown}
      onMouseLeave={onMouseUp}
      onMouseUp={onMouseUp}
      onClick={onReset}
    >
      {mouseDown ? 'ðŸ˜¯' : 'ðŸ™‚'}
    </Button>
  );
});

Reset.displayName = 'Reset';

const Button = styled.button`
  font-size: 1.5vw;
  cursor: pointer;
  font-weight: 700;
  border-width: 0.15vw;
  border-style: solid;
  background-color: #d1d1d1;
  border-color: white #9e9e9e #9e9e9e white;
`;
