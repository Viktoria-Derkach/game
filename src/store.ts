import { configureStore } from '@reduxjs/toolkit';

import { reducer } from '@/modules/GameWithRedux';

export const store = configureStore({
  reducer: {
    game: reducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
