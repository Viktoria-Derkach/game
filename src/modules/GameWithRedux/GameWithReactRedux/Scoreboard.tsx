import React, { FC, useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { GameLevels, LevelNames } from '@/modules/GameSettings';
import { RootState } from '@/store';
import { Scoreboard as ScoreboardComponent } from '@/components/Scoreboard';
import { actions } from '@/modules/GameWithRedux';

export const Scoreboard: FC = () => {
  const { level, time, bombs, flagCounter } = useSelector(
    ({ game: { level, time, bombs, flagCounter } }: RootState) => ({
      level,
      time,
      bombs,
      flagCounter,
    })
  );

  const dispatch = useDispatch();

  const onChangeLevel = useCallback(
    ({ target: { value: level } }: React.ChangeEvent<HTMLSelectElement>) =>
      dispatch(actions.changeLevel(level as LevelNames)),
    // Stryker disable next-line ArrayDeclaration
    []
  );

  const onReset = useCallback(
    () => dispatch(actions.reset()),
    // Stryker disable next-line ArrayDeclaration
    []
  );

  return (
    <ScoreboardComponent
      time={String(time)}
      bombs={String(bombs - flagCounter)}
      levels={GameLevels as unknown as string[]}
      defaultLevel={level}
      onChangeLevel={onChangeLevel}
      onReset={onReset}
    />
  );
};
