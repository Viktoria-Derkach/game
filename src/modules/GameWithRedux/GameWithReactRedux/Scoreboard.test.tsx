import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import { store } from '@/store';

import { Scoreboard } from './Scoreboard';

it('Scoreboard renders correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Scoreboard />
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
