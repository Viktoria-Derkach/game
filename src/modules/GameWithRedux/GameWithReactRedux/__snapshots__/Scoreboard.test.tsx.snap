// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard test cases Scoreboard check 1`] = `
<DocumentFragment>
  <div
    class="css-1gexv9m"
  >
    <div
      class="css-d2q1pa"
    >
      0
    </div>
    <select
      class="css-55qvia"
    >
      <option
        class="css-zcmdxr"
        value="beginner"
      >
        beginner
      </option>
      <option
        class="css-zcmdxr"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class="css-zcmdxr"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-1bva717"
    >
      ðŸ™‚
    </button>
    <div
      class="css-d2q1pa"
    >
      7
    </div>
  </div>
</DocumentFragment>
`;

exports[`Scoreboard test cases Scoreboard check 2`] = `
<DocumentFragment>
  <div
    class="css-1gexv9m"
  >
    <div
      class="css-d2q1pa"
    >
      0
    </div>
    <select
      class="css-55qvia"
    >
      <option
        class="css-zcmdxr"
        value="beginner"
      >
        beginner
      </option>
      <option
        class="css-zcmdxr"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class="css-zcmdxr"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-1bva717"
    >
      ðŸ™‚
    </button>
    <div
      class="css-d2q1pa"
    >
      7
    </div>
  </div>
</DocumentFragment>
`;
